### PR TITLE
dug: full resolve option

### PR DESCRIPTION
**Pull request by khibino**
add `--full` option to dug command.

### CODE REVIEW

**Comment by khibino** [comment]
* `dug --full iij.ad.jp. A`
* `dug --full www.aol.com. A`

File: dnsext-full-resolver/DNS/Cache/Log.hs [hunk 1]
```diff
@@ -48,15 +49,21 @@ outputHandle o = case o of
   Stdout  ->  stdout
   Stderr  ->  stderr
 
-new :: Handle -> Level -> IO (IO (), Level -> [String] -> IO (), IO (Int, Int))
+new :: Handle -> Level -> IO (IO (), Level -> [String] -> IO (), IO (Int, Int), IO ())
```
Comment (kazu-yamamoto) [inline dnsext-full-resolver/DNS/Cache/Log.hs:52]: Would you define type synonyms for each IO action such as `Flush`?

File: dnsext-full-resolver/dug/dug.hs [hunk 2]
```diff
@@ -25,13 +28,21 @@ main = do
...
         case mHostTyp of
           Nothing -> putStr help
-          Just (host,typ) -> do
-              ex <- operate mserver host typ ctl
-              case ex of
-                Left err -> fail $ show err
-                Right rs -> putStr $ pprResult rs
+          Just (host,typ)
+            | full      -> do
+                ex <- fullResolve disableV6NS Log.Stdout Log.INFO host typ
```
Comment (kazu-yamamoto) [inline dnsext-full-resolver/dug/dug.hs:37]: Why `disableV6NS`?

File: dnsext-full-resolver/DNS/Cache/Log.hs (hunk 1 above)
Comment (khibino) [reply dnsext-full-resolver/DNS/Cache/Log.hs]: https://github.com/kazu-yamamoto/dnsext/pull/17/commits/258244b20181f1e663f80997dc586341678a56d8

**Review by kazu-yamamoto — CHANGES_REQUESTED** [review]
I said "each action". 

So, `IO (IO (), Level -> [String] -> IO (), IO (Int, Int), Flush)` should be
`IO (Foo, Bar, Baz, Flush)`

**Review by kazu-yamamoto — APPROVED** [review]
LGTM

**Comment by kazu-yamamoto** [comment]
Would you rebase and merge this PR by yourself?

**Comment by khibino** [comment]
Rebased and merged. Thanks!

**Pull request merged**